tsuchinoko/execution/bluesky_adaptive.py

```diff
@@ -0,0 +1,141 @@
+            except zmq.ZMQError:
+                break
+            else:
+                assert 'target_measured' in payload
+                x, y = payload['target_measured']
+                # TODO: Its highly recommended to extract a variance for y; we might piggyback on y,
+                #       s.t. y = [y1, y2, ..., yn, y1variance, y2variance, ..., ynvariance]
+                # TODO: Any additional quantities to be interrogated in Tsuchinoko can be included in the trailing dict
+                new_measurements.append((x, y, 1, {}))
+                # stash the last position measured as the 'current' position of the instrument
+                self.position = x
+        return new_measurements
+
+    def get_position(self) -> Tuple:
+        # return last measurement position received from bluesky-adaptive
+        return self.position
+
+    def send_payload(self, payload: dict):
+        logger.info(f'message: {payload}')
+        self.socket.send(pickle.dumps(payload))
+
+    def recv_payload(self, flags=0) -> dict:
+        payload_response = pickle.loads(self.socket.recv(flags=flags))
+        logger.info(f'response: {payload_response}')
+        return payload_response
+
+
+# ----------------------------------------------------------------------------------------------------------------------
+# This is a prototype Agent to be used with bluesky-adaptive. This should be extracted before merge.
+
+from bluesky_adaptive.agents.base import Agent
+
+
+class TsuchinokoAgent(Agent):
+    def __init__(self, *args, host='127.0.0.1', port=5557, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.host = host
+        self.port = port
+        self.outbound_measurements = []
+        self.setup_socket()
+
+    def setup_socket(self):
+        self.context = zmq.Context()
+        self.socket = self.context.socket(zmq.PAIR)
+
+        # Attempt to connect, retry every second if fails
+        while True:
+            try:
+                self.socket.connect(f"tcp://{self.host}:{self.port}")
+            except zmq.ZMQError:
+                logger.info(f'Unable to connect to tcp://{self.host}:{self.port}. Retrying in 1 second...')
+                time.sleep(1)
+            else:
+                logger.info(f'Connected to tcp://{self.host}:{self.port}.')
+                break
+
+        # Limit number of buffered messages to 1; dumps any earlier targets if new ones come in before payloads are received
+        self.socket.setsockopt(zmq.CONFLATE, 1)
+
+    def tell(self, x, y):
+        # Send measurement to BlueskyAdaptiveEngine
+        payload = {'target_measured': (x, y)}
+        self.send_payload(payload)
+
+    def ask(self, batch_size: int) -> Tuple[Sequence[Dict[str, ArrayLike]], Sequence[ArrayLike]]:
+        # Get targets from BlueskyAdaptiveEngine
+        payload = self.recv_payload()
+        assert 'targets' in payload
+        return [{}], payload['targets']
```

Comment:
This is fine, but if there is any info you'd want to appear in Tiled/Databroker after the experiment, this is the place to put it.